datafusion/core/src/execution/disk_manager.rs

```diff
@@ -103,21 +103,26 @@ impl DiskManager {
     }
 
     /// Return a temporary file from a randomized choice in the configured locations
-    pub fn create_tmp_file(&self) -> Result<NamedTempFile> {
+    ///
+    /// If the file can not be created for some reason, returns an
+    /// error message referencing the request description
+    pub fn create_tmp_file(&self, request_description: &str) -> Result<NamedTempFile> {
```

Comment:
Hopefully once we rework the error handling story, we can move away from this nonsense